expression/builtin_time.go

```diff
@@ -1257,37 +1259,128 @@ type strToDateFunctionClass struct {
 	baseFunctionClass
 }
 
-func (c *strToDateFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+func (c *strToDateFunctionClass) getRetTp(arg Expression, ctx context.Context) (tp byte, fsp int) {
+	tp = mysql.TypeDatetime
+	if _, ok := arg.(*Constant); !ok {
+		return tp, types.MaxFsp
+	}
+	strArg := WrapWithCastAsString(arg, ctx)
+	format, isNull, err := strArg.EvalString(nil, ctx.GetSessionVars().StmtCtx)
+	if err != nil || isNull {
+		return
+	}
+	isDuration, isDate := types.GetFormatType(format)
+	if isDuration && !isDate {
+		tp = mysql.TypeDuration
+	} else if !isDuration && isDate {
+		tp = mysql.TypeDate
+	}
```

Comment:
what if `!isDuration && !isDate` and `isDuration && isDate` ?

Reply:
both return mysql.TypeDatetime